examples/format_baristaseq.py

```diff
@@ -0,0 +1,132 @@
+"""
+This script contains TileFetchers and FetchedTile objects to convert data donated by the Zador lab
+to SpaceTx format. The locations of the data files for use with this script can be found in the
+docstring for `format_data`
+"""
+
+import os
+from typing import Callable, Mapping, Tuple, Union
+
+import click
+import numpy as np
+from skimage.io import imread
+from slicedimage import ImageFormat
+
+from starfish.experiment.builder import (FetchedTile, TileFetcher,
+                                         write_experiment_json)
+from starfish.types import Coordinates, Indices, Number
+
+DEFAULT_TILE_SHAPE = 1000, 800
+
+
+class BaristaSeqTile(FetchedTile):
+    def __init__(self, file_path):
+        self.file_path = file_path
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        return DEFAULT_TILE_SHAPE
+
+    @property
+    def coordinates(self) -> Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]]:
+        # these are dummy coordinates
+        return {
+            Coordinates.X: (0.0, 0.0001),
+            Coordinates.Y: (0.0, 0.0001),
+            Coordinates.Z: (0.0, 0.0001),
+        }
+
+    @property
+    def format(self) -> ImageFormat:
+        return ImageFormat.TIFF
+
+    @property
+    def tile_data(self) -> Callable[[], np.ndarray]:
+        def future():
+            return imread(self.file_path)
+        return future
+
+
+class BaristaSeqTileFetcher(TileFetcher):
+    def __init__(self, input_dir) -> None:
+        self.input_dir = input_dir
+
+    def get_tile(self, fov: int, hyb: int, ch: int, z: int) -> FetchedTile:
+        subdir = 'primary'
+        round_dir = f'r{hyb}'
+        if hyb == 0:
+            filename = f'T{fov+1:05}C{ch+1:02}Z{z+1:03}.tif'
+        else:
+            filename = f'alignedT{fov+1:05}C{ch+1:02}Z{z+1:03}.tif'
+        file_path = os.path.join(self.input_dir, subdir, round_dir, filename)
+        return BaristaSeqTile(file_path)
+
+
+class BaristaSeqNucleiTileFetcher(TileFetcher):
+    def __init__(self, input_dir, aux_type) -> None:
+        self.input_dir = input_dir
+
+    def get_tile(self, fov: int, hyb: int, ch: int, z: int) -> FetchedTile:
+        subdir = 'nissl'
+        filename = f'T00001C05Z{z+1:03}.tif'
```

Comment:
will these ever change?

Reply:
The file names? No, not for this dataset. But generally, all TileFetcher code is one-off. It's worthless if the user structures their directory differently or changes file naming procedures for _future datasets_.